osm2streets/src/transform/classify_intersections.rs

```diff
@@ -1,85 +1,229 @@
 use crate::osm::NodeID;
+use crate::types::Movement;
+use crate::Direction;
 use crate::IntersectionComplexity::*;
-use crate::{Direction, IntersectionComplexity, StreetNetwork};
+use crate::{
+    ConflictType, DrivingSide, IntersectionComplexity, RestrictionType, Road, StreetNetwork,
+};
+use std::cmp::{max, min};
+use std::collections::BTreeMap;
 
 /// Determines the initial complexity of all intersections. Intersections marked "Crossing" are
 /// considered "unclassified" and will be updated with a guess, others will be left unchanged.
 pub fn classify_intersections(streets: &mut StreetNetwork) {
-    let mut changes: Vec<(NodeID, IntersectionComplexity)> = Vec::new();
+    let mut changes: Vec<_> = Vec::new();
     for (id, inter) in &streets.intersections {
         if let Crossing = inter.complexity {
```

Comment:
```suggestion
        if inter.complexity == Crossing {
```

Reply:
I was fighting with rust syntax a while a go and had convinced myself that this wasn't valid for enums. I'm glad it is!

Reply:
If the enum derives or implements `PartialEq`, then you can just do regular `==`. `if let` is nice for extracting something from an enum case. Like if we later do... `if let IntersectionComplexity::Crossing(conflict_type) = inter.complexity`

Reply:
> If the enum derives or implements `PartialEq` ...

This is the details that I didn't realise was relevant! Makes sense now.